### PR TITLE
Add nuget dependency

### LINKED ISSUE #15536: [Build] make ios dylib erro
### Describe the issue

follow the docs build iOS, but error with
FileNotFoundError: [Errno 2] No such file or directory: 'nuget'

### Urgency

_No response_

### Target platform

iOS

### Build script

./build.sh --config Release --use_xcode --ios --ios_sysroot iphoneos --osx_arch arm64

### Error / output

FileNotFoundError: [Errno 2] No such file or directory: 'nuget'

### Visual Studio Version

_No response_

### GCC / Compiler Version

_No response_

### PR DESCRIPTION
**Pull request by natke**
### Description
Add nuget dependency to iOS buid

### Motivation and Context
Fixes #15536

### CODE REVIEW

File: docs/build/ios.md [hunk 1]
```diff
@@ -42,6 +42,7 @@ Follow the instructions below to build ONNX Runtime for iOS.
 * Xcode, https://developer.apple.com/xcode/
 * CMake, https://cmake.org/download/
 * Python 3, https://www.python.org/downloads/mac-osx/
+* Nuget https://formulae.brew.sh/formula/nuget (used to download the protobuf compiler, which is needed to build ONNX Runtime)
```
Comment (edgchen1) [inline docs/build/ios.md:45]: should we link to the MS instructions instead?
https://learn.microsoft.com/en-us/nuget/install-nuget-client-tools#macoslinux

Comment (natke) [reply docs/build/ios.md:45]: I looked at those but found them a little confusing. I thought a single step would be better. Let me know your thoughts

Comment (edgchen1) [reply docs/build/ios.md:45]: suggested that because it seems like a more official source.

**Comment by edgchen1** [comment]
FYI this PR removes the nuget build dependency.
https://github.com/microsoft/onnxruntime/pull/15731

**Pull request closed without merging**